Troubleshooting: add an article about gray boxes in grids (#5131)
DevExpress/devextreme-documentation: concepts/80 Troubleshooting/05 DataGrid and TreeList Issues/10 Gray Boxes Appear After Data Loading/00 Gray Boxes Appear After Data Loading.md

> In certain cases, gray boxes may remain visible after DataGrid or TreeList finishes data loading:

Reviewer: ```suggestion
In certain cases, gray boxes may remain visible after DataGrid or TreeList finishes loading data:
```

> You may experience an issue, when DataGrid or TreeList finishes data loading, gray boxes are still visible:

Reviewer 2: > You may experience an issue, when DataGrid or TreeList finishes data loading, gray boxes are still visible:

There is something fundamentally wrong with punctuation and flow order in this sentence...

```suggestion
In certain cases, gray boxes may remain visible after DataGrid or TreeList finishes data loading:
```

![DataGrid with Gray Boxes Visible](/images/DataGrid/datagrid-gray-boxes.png)

> These gray boxes are row placeholders that appear when the component takes time to load and render rows. These placeholders can remain visible due to the following reasons:

Reviewer: ```suggestion
These gray boxes are row placeholders that appear while the component loads and renders rows. These placeholders can remain visible in the following cases:
```

> These gray boxes are row placeholders that appear when the component takes time to load and render rows. The issue described above occurs due to the following reasons:

Reviewer 2: ```suggestion
These gray boxes are row placeholders that appear when the component takes time to load and render rows. These placeholders can remain visible due to the following reasons:
```

> - The component's size does not match the container size.

Reviewer 2: ```suggestion
- The component size does not match the container size.
```

> - The component's height property is not consistent with the space the component occupies.

Reviewer 2: 1. If you speak of a property, shouldn't it be a link?
2. What do you mean by "the height property is not consistent"? I do not understand what it means. Please clarify.

Reviewer 2: This issue has been addressed.

- An error occurs during data loading.

> - The loaded data is incorrect (has a wrong structure, missing required fields, the number of data items is not expected, etc.).

Reviewer 2: 1. Yes, you probably can emphasize "loaded data" with a definite article in this scenario. However, I would have refrained from it.
2. What do you mean by "the number of data items in not expected"? What is it? When it may happen and why? And, finally, what are "data items"? Dataset records? Rows? Or individual cells? I've got more questions than answers here.
3. If you use etc. with a closing bracket at the end of a sentence, you do not need a second period after the bracket: "etc.)" is more than enough.

I cannot rewrite this sentence for you due to the lack of knowledge and context, therefore, please clarify.

Reviewer 2: This issue has been addressed.

> - Unexpected external interactions (rerendering, resizing, state storing, etc.).

Reviewer 2: Resizing what? Browser viewport/window? I'm unsure I understand "state storing"... but let it be.

Again we have "etc." with two periods and a bracket. Only one period before the closing bracket is enough. Like this:

> etc.)

Again, I cannot rewrite this line for your, so please clarify.

Reviewer 2: This issue has been addressed.

> - Unexpected external interactions (the component is re-rendering, the container has been resized, saved component state is corrupted, etc.)

Reviewer 2: First. We should stick to the same tense. A simple one, preferably.

Second. "state" is countable, which requires an article from us.

```suggestion
- Unexpected external interactions (when the component is re-rendering, the container is resized, the saved component state is corrupted, etc.)
```
